### PR TITLE
feat: allow rollup update

### PR DESCRIPTION
**Pull request by gzeoneth**
This PR included 2 changes that should only affect privileged methods

- [x] Allow rollup admin to update rollup addresses in Bridge contracts, and synchronizing those changes to Outbox, RollupEventInbox, and Sequencer Inbox. These changes are audited as part of BOLD in https://github.com/OffchainLabs/bold/pull/340 but for proxy admin instead of rollup admin. Allowing rollup admin to call those function make the upgrade executor being able to call those function easier (or otherwise would require a upgradeAndCall via the ProxyAdmin contract).

- [x] Introduce `whenNotPausedOrDeprecated` modifier, which allow validator to withdraw their stake when the rollup is deprecated by changing the rollup address in the bridge to something else

Superseding
* https://github.com/OffchainLabs/nitro-contracts/pull/54
* https://github.com/OffchainLabs/nitro-contracts/pull/48

### CODE REVIEW

File: src/rollup/RollupUserLogic.sol [hunk 1]
```diff
@@ -227,14 +227,24 @@ abstract contract AbsRollupUserLogic is
         stakeOnNode(msg.sender, latestNodeCreated());
     }
 
+    modifier whenNotPausedOrDeprecated() {
```
Comment (yahgwai) [inline src/rollup/RollupUserLogic.sol:230]: Maybe move this modifier to the top of the contract

**Review by gvladika — DISMISSED** [review]
lgtm, might be useful to emit RollupUpdated in Bridge

File: test/contract/arbRollup.spec.ts [hunk 2]
```diff
@@ -789,6 +802,44 @@ describe('ArbRollup', () => {
     await rollup.confirmNextNode(challengerNode)
   })
 
+  it('allow force refund staker with pending node', async function () {
+    await rollupAdmin.connect(await impersonateAccount(upgradeExecutor)).pause()
```
Comment (yahgwai) [inline test/contract/arbRollup.spec.ts:806]: missing .wait() - and on some other transactions below

**Review by yahgwai — APPROVED** [review]

File: test/contract/arbRollup.spec.ts (hunk 2 above)
Comment (gzeoneth) [reply test/contract/arbRollup.spec.ts:806]: that should not be necessary right? otherwise the test should be failing

**Review by yahgwai — APPROVED** [review]

File: test/contract/arbRollup.spec.ts (hunk 2 above)
Comment (yahgwai) [reply test/contract/arbRollup.spec.ts:806]: could just be getting lucky with timing. You might get intermittent failures with this

**Pull request merged**